parse_trending_models.py

```diff
@@ -1,65 +1,350 @@
-from huggingface_hub import HfApi
-import os
-from datetime import datetime
+import argparse
+from huggingface_hub import HfApi, file_exists, upload_file
 import json
+from datetime import datetime
+import requests
+import os
+from dotenv import load_dotenv
+import logging
+from typing import Dict, List, Any, Optional
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+)
+logger = logging.getLogger(__name__)
+
+load_dotenv()
+
+AVOCADO_AUTHORS = [
+    "ariG23498",
+    "reach-vb",
+    "pcuenq",
+    "burtenshaw",
+    "dylanebert",
+    "davanstrien",
+    "merve",
+    "sergiopaniego",
+    "Steveeeeeeen",
+    "ThomasSimonini",
+]
+
+
+class ModelChecker:
+    def __init__(self, debug: bool = False, limit: int = 100):
+        self.debug = debug
+        self.limit = limit
+
+        self.hf_api = HfApi()
+        self.data_folder = "trending_data"
+        self.repo_id = "ariG23498/trending_models"
+        self.slack_webhook_url = os.getenv("SLACK_WEBHOOK_URL")
+        self.today = datetime.now().strftime("%Y-%m-%d")
+
+        os.makedirs(self.data_folder, exist_ok=True)
+        self.output_file = f"{self.data_folder}/{self.today}.json"
+
+        self.problematic_models = {
+            "models_with_no_library_name": [],
+            "models_with_no_pipeline_tag": [],
+            "models_with_custom_code": [],
+            "models_with_no_discussion_tab": [],
```

Comment:
we can skip this check - I don't think it provides any value

Reply:
I was using this in order to skip from looking at the metadata. The alert has it as a side effect.

Reply:
okay